tests/ImageSharp.Tests/PixelFormats/PixelOperations/PixelOperationsTests.cs

```diff
@@ -1177,14 +1152,22 @@ public void Verify()
                 {
                     Span<Vector4> expected = MemoryMarshal.Cast<TDest, Vector4>(this.ExpectedDestBuffer.AsSpan());
                     Span<Vector4> actual = MemoryMarshal.Cast<TDest, Vector4>(this.ActualDestBuffer.GetSpan());
+                    var comparer = new ApproximateFloatComparer(TestEnvironment.Is64BitProcess ? 0.0001F : 0.001F);
 
-                    var comparer = new ApproximateFloatComparer(0.001f);
                     for (int i = 0; i < count; i++)
                     {
-                        // ReSharper disable PossibleNullReferenceException
                         Assert.Equal(expected[i], actual[i], comparer);
+                    }
+                }
+                else if (typeof(IPixel).IsAssignableFrom(typeof(TDest)))
+                {
+                    Span<TDest> expected = this.ExpectedDestBuffer.AsSpan();
+                    Span<TDest> actual = this.ActualDestBuffer.GetSpan();
+                    var comparer = new ApproximateFloatComparer(TestEnvironment.Is64BitProcess ? 0.0001F : 0.001F);
 
-                        // ReSharper restore PossibleNullReferenceException
+                    for (int i = 0; i < count; i++)
+                    {
+                        Assert.Equal((IPixel)expected[i], (IPixel)actual[i], comparer);
```

Comment:
I don't really understand this path.

How can [`ApproximateFloatComparer`](https://github.com/SixLabors/ImageSharp/blob/a1784a67bf492b1b02e99655a39aa5e5acf81988/tests/ImageSharp.Tests/TestUtilities/ApproximateFloatComparer.cs) compare `IPixel`-s?

If I understand your intent right, you wanted to loosen the strictness for the cases where `TDest` is a pixel (almost everything). I don't think it's always a good idea. For example `Rgb32 -> Rgb24` conversion should be always exact.

Reply:
They will be exact at that bit depth since our approach uses 16bits per component. I noticed very minor fluctuations for Short4 and HalfVector4 though in one of the longer array runs. Small enough to disappear but large enough to fail the exact comparer.

Reply:
When we attach more metadata to a TPixel type we could be more explicit

Reply:
I missed that you added `ApproximateFloatComparer : IEqualityComparer<IPixel>` in my previous comment.

Still somewhat concerned about tolerant comparison in cases we don't need it. I would add a cheap heuristic approach like this:
- To take a default-valued parameter `preferExactComparison = true` on `TestOperation` and pass it forward to `Verify`
- Pass `false` in `FromCompandedScaledVector4` and similar
- Change the condition:

```C#
else if (typeof(IPixel).IsAssignableFrom(typeof(TDest)) && !preferExactComparison && (IsComplexPixel<TSource>() || IsComplexPixel<TDest>()))
```

and 

```C#
static bool IsComplexPixel<TPixel>() => Unsafe.SizeOf<TPixel>() > sizeof(int);
```

Reply:
```
static bool IsComplexPixel<TPixel>() => Unsafe.SizeOf<TPixel>() > sizeof(int);
```

Nearly... There's a few types that would fail this test. I've written a version that explicitly captured them. We really need a `PixelTypeInfo.BitsPerComponent` property.